centreon/src/App/ActivityLogging/Domain/Event/LogActivityEventHandler.php

```diff
@@ -49,26 +49,39 @@ public function __construct(
 
     public function __invoke(AggregateCreated|AggregateUpdated|AggregateDeleted $event): void
     {
-        if (! $this->activityLogFactories->has($event->aggregate::class)) {
-            throw new \LogicException(\sprintf('There is no "%s" for "%s", did you add a service with "activity_logging.activity_log_factory" tag?', ActivityLogFactoryInterface::class, $event->aggregate::class));
+        $aggregates = is_array($event->aggregate) ? $event->aggregate : [$event->aggregate];
+
+        if ($aggregates === []) {
+            return;
+        }
+
+        if (! $this->activityLogFactories->has($aggregates[0]::class)) {
+            throw new \LogicException(\sprintf('There is no "%s" for "%s", did you add a service with "activity_logging.activity_log_factory" tag?', ActivityLogFactoryInterface::class, $aggregates[0]::class));
         }
 
         /** @var ActivityLogFactoryInterface<AggregateRoot<AggregateRootId>> $factory */
-        $factory = $this->activityLogFactories->get($event->aggregate::class);
+        $factory = $this->activityLogFactories->get($aggregates[0]::class);
 
         $action = match (true) {
             $event instanceof AggregateCreated => ActionEnum::Add,
             $event instanceof AggregateUpdated => ActionEnum::Update,
             $event instanceof AggregateDeleted => ActionEnum::Delete,
         };
 
-        $activityLog = $factory->create(
-            action: $action,
-            aggregate: $event->aggregate,
-            firedBy: new Actor(id: new ActorId($event->creatorId)),
-            firedAt: $event->firedAt(),
-        );
+        $activityLogs = [];
+        foreach ($aggregates as $aggregate) {
+            $activityLogs[] = $factory->create(
+                action: $action,
+                aggregate: $aggregate,
+                firedBy: new Actor(id: new ActorId($event->creatorId)),
+                firedAt: $event->firedAt(),
+            );
+        }
 
-        $this->repository->add($activityLog);
+        if (count($activityLogs) === 1) {
+            $this->repository->add($activityLogs[0]);
+        } else {
+            $this->repository->addMultiple($activityLogs);
+        }
```

Comment:
With my previous comment, it'll be:
```suggestion
        $this->repository->add(...$activityLogs);
```
Easier 😉